upgrade/upgrade_provider.go

```diff
@@ -63,7 +63,13 @@ func UpgradeProvider(ctx context.Context, repoOrg, repoName string) (err error)
 
 	err = stepv2.PipelineCtx(ctx, "Discover Provider", func(ctx context.Context) {
 		repo.root = OrgProviderRepos(ctx, repoOrg, repoName)
-		repo.defaultBranch = pullDefaultBranch(ctx, "origin")
+		// If the user set --repo-path as CWD, assume all git content is already in-place; simply infer the main
+		// branch without pulling anything. Otherwise, pull.
+		if GetContext(ctx).repoPath == "." {
```

Comment:
I think we should make this absolute. Otherwise `upgrade-provider .` will be treated differently then `upgrade-provider $(pwd)`.
```suggestion
		if filepath.Abs(GetContext(ctx).repoPath) == filepath.Abs(".") {
```

Reply:
Yeah I can do that. One sec.